src/main/java/org/commcare/formplayer/services/MediaHandler.kt

```diff
@@ -19,16 +21,37 @@ class MediaHandler(val file: MultipartFile) {
     /**
      * Saves file in the given parent directory
      */
-    fun saveFile(parentDirPath: Path): String {
+    fun saveFile(parentDirPath: Path, session: SerializableFormSession, username: String, asUser: String,
+                 domain: String, appId: String): String {
         validateFile()
         val fileId = PropertyUtils.genUUID()
         val parent = parentDirPath.toFile()
         parent.mkdirs()
         var fileIdWithExt = fileId
-        FileUtils.getExtension(file.originalFilename)?.let { fileIdWithExt = "$fileId.$it" }
-        val desintationFile = getMediaFilePath(parentDirPath, fileIdWithExt).toFile()
+        val fileExtension = FileUtils.getExtension(file.originalFilename)
+        fileExtension?.let { fileIdWithExt = "$fileId.$it" }
+        val filePath = getMediaFilePath(parentDirPath, fileIdWithExt)
+        val destinationFile = filePath.toFile()
+
+        val metaDataRecordToDelete = MediaMetaDataService().findByFilePath(filePath.toString())
+        val mediaMetaDataService = MediaMetaDataService()
+        metaDataRecordToDelete?.let{ mediaMetaDataService.deleteMetaDataById(it.id) }
```

Comment:
Also curious if there is a reason behind doing a `exists` check before deleting the record here ?  I think it should be fine to call delete directly and probably put this in a try..catch block catching all exceptions as we don't care if delete record fails. 

````
try {
  mediaMetaDataService.deleteMetaDataById(fileId)
}catch(e: Exception){
 // ignore, we don't want to crash even if delete fails
}